test/test_library/unit/test_init.py

```diff
@@ -0,0 +1,1858 @@
+        self.sqlite_db.create_tables()
+        lib = Library(self.cache, self.sqlite_db)
+
+        ds1 = DatasetFactory()
+        table1 = TableFactory(dataset=ds1)
+
+        ds2 = DatasetFactory()
+        table2 = TableFactory(dataset=ds2)
+        tables = lib.tables_no_columns
+        self.assertIn(table1, tables)
+        self.assertIn(table2, tables)
+
+    # .table tests
+    def test_returns_table_found_by_vid(self):
+        self.sqlite_db.create_tables()
+        lib = Library(self.cache, self.sqlite_db)
+
+        ds1 = DatasetFactory()
+        table1 = TableFactory(dataset=ds1)
+
+        self.assertEquals(lib.table(table1.vid), table1)
+
+    def test_returns_table_found_by_id(self):
+        self.sqlite_db.create_tables()
+        lib = Library(self.cache, self.sqlite_db)
+
+        ds1 = DatasetFactory()
+        table1 = TableFactory(dataset=ds1)
+        assert table1.id_ != table1.vid
+
+        self.assertEquals(lib.table(table1.id_), table1)
+
+    def test_raises_NotFoundError_if_table_not_found(self):
+        self.sqlite_db.create_tables()
+        lib = Library(self.cache, self.sqlite_db)
+
+        # testing
+        with self.assertRaises(NotFoundError):
+            lib.table('vid')
+
+    # .column tests
+    def test_finds_column_by_vid(self):
+        lib = Library(self.cache, self.sqlite_db)
+        column1 = ColumnFactory()
+
+        # testing
+        ret = lib.column(column1.vid)
+        self.assertEquals(column1.id_, ret.id_)
+        self.assertEquals(column1.vid, ret.vid)
+        self.assertEquals(column1.table, ret.table)
+
+    def test_finds_column_by_id(self):
+        lib = Library(self.cache, self.sqlite_db)
+        column1 = ColumnFactory()
+
+        # testing
+        ret = lib.column(column1.id_)
+        self.assertEquals(column1.id_, ret.id_)
+        self.assertEquals(column1.vid, ret.vid)
+        self.assertEquals(column1.table, ret.table)
+
+    def test_raises_NotFoundError_if_column_not_found(self):
+        lib = Library(self.cache, self.sqlite_db)
+
+        # testing
+        with self.assertRaises(NotFoundError):
+            lib.column('the-vid')
+
+    # .derived_tables tests
+    def test_returns_tables_found_by_proto_vid(self):
+        # prepare state
+        self.sqlite_db.create_tables()
+        lib = Library(self.cache, self.sqlite_db)
+
+        ds1 = DatasetFactory()
+        ds2 = DatasetFactory()
+        ds3 = DatasetFactory()
+        table1 = TableFactory(dataset=ds1, proto_vid='1')
+        table2 = TableFactory(dataset=ds2, proto_vid='1')
+        table3 = TableFactory(dataset=ds3, proto_vid='2')
+
+        # testing
+        derived_tables = lib.derived_tables('1')
+        self.assertEquals(len(derived_tables), 2)
+        self.assertIn(table1, derived_tables)
+        self.assertIn(table2, derived_tables)
+        self.assertNotIn(table3, derived_tables)
+
+    @unittest.skip('.all() method of query never raises NoResultFound error.')
```

Comment:
Hmm .. not sure what the correct behavior is, though I suspect it is to return and empty list when there are no partitions.